analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsNull.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors
+{
+    internal static class IsNull
+    {
+        public static ProgramState Process(SymbolicContext context, IIsNullOperationWrapper isPattern) =>
```

Comment:
Can you add a comment about "IsNullOperation"? It is totally unclear to me (and probably any future reader) what it does. It seems related to `??` and `??=` but unrelated to `?.`, `?:`, and `is null`. This is especially confusing because there are also ICoalesceAssignmentOperationWrapper and ICoalesceOperationWrapper which are "binary" (e.g. `Value` and `WhenNull`) but this one only has `Operand` so it is unary.

Reply:
From the `IsNull` name, it's pretty clear what it does.
It's unclear when precisely Roslyn generates those. And this is documented via test cases.